### ISSUE TITLE
Custom descriptions not work

### ISSUE THREAD

**Issue by aa1ss2** [nushell/nushell#8825]
### Describe the bug

Custom descriptions return error type ,missing string identifier.

### How to reproduce

1. book example [https://www.nushell.sh/book/custom_completions.html#custom-completion-and-extern](url)
2. define
```nushell
def my_commits [] {
    [
        { value: "5c2464", description: "Add .gitignore" },
        { value: "f3a377", description: "Initial commit" }
    ]
}
```
3. define
```nushell
def my-command [commit: string@my_commits] {
    print $commit
}
```

### Expected behavior

Return the correct type

### Screenshots

![1](https://user-images.githubusercontent.com/79680252/230761967-0e7ebf92-afbb-4985-95fd-fde1bfe699cd.png)

### Configuration

| key                | value                                                                                                                                                                                                     |
| ------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| version            | 0.78.1                                                                                                                                                                                                    |
| branch             | main                                                                                                                                                                                                      |
| commit_hash        | 637283ffad895358cca644223f83eab23123db4a                                                                                                                                                                  |
| build_os           | windows-x86_64                                                                                                                                                                                            |
| build_target       | x86_64-pc-windows-msvc                                                                                                                                                                                    |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)                                                                                                                                                                       |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc                                                                                                                                                                             |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)                                                                                                                                                                       |
| build_time         | 2023-04-09 09:55:02 +08:00                                                                                                                                                                                |
| build_rust_channel | release                                                                                                                                                                                                   |
| features           | dataframe, default, sqlite, zip                                                                                                                                                                           |
| installed_plugins  | custom-value generate, custom-value generate2, custom-value update, from eml, from ics, from ini, from vcf, gstat, inc, nu-example-1, nu-example-2, nu-example-3, query, query json, query web, query xml |

**Comment by amtoine** [comment]
@aa1ss2 

i think this has been introduced in #8635 and i can reproduce it the latest `main` :+1: 

a correct call would be
```bash
my-command "5c2464"
```
and not
```bash
my-command 5c2464
```
as given by the auto-completion :thinking:

**Comment by amtoine** [comment]
even worse, if you try to tab-complete on `f` it works as expected, but nothing happens on `5` :thinking:

**Comment by aa1ss2** [comment]
Thanks for the answer,i know correct call is `my-command "5c2464"` ,the problem is i input `my-command "` and press tab 
 nothing happens ,If don't enter double quotes ,return wrong value ,this is very troubling for me

**Comment by amtoine** [comment]
that looks like a bug to me :+1:

**Comment by 1Kinoti** [comment]
i believe this has been fixed by #8845.

**Comment by amtoine** [comment]
good catch @1Kinoti, i cannot reproduce with the latest revision of Nushell :ok_hand: 
will be closing this for now :relieved:
